core/src/main/java/org/infinispan/interceptors/distribution/L1NonTxInterceptor.java

```diff
@@ -301,11 +317,15 @@ private BasicInvocationStage handleDataWriteCommand(InvocationContext ctx, DataW
       });
    }
 
-   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue)
-         throws Throwable {
+   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue) {
       if (ctx.isOriginLocal() && !cdl.localNodeIsOwner(command.getKey())) {
+         CompletableFuture<?> pendingAcks = commandAckCollector.getCollectorCompletableFuture(command.getCommandInvocationId(), false);
```

Comment:
the TriangleInterceptor is placed before the L1 and the L1 needs to send the invalidation after the operation is completed. 
I can change the name if you like, but I think you got confused by something else, no?